### PR TITLE
Some ringdown updates

### PR DESCRIPTION
**Pull request by cdcapano**
This adds final mass and spin from `f0`, `tau` functions to conversions. These functions use the Berti et al. fits. Currently, only support for l = m = 2 is provided.

A `qnm` module is also added to `distributions`, which includes a `UniformF0Tau` distribution. This distribution allows for constraints on final mass and spin to be provided.

Both of these are largely copied from @micamu area theorem branch.

Note: strictly speaking, the `UniformF0Tau` distribution isn't needed, since the same thing could be accomplished by providing a `Uniform` distribution (for f0 and tau) along with constraints (for final mass and spin) to the `JointDistribution`. That's a bit complicated though. `UnformF0Tau` is a convenience class, similar to `UniformSky`.

### CODE REVIEW

**Comment by ahnitz** [comment]
@micamu Poke, status?

**Comment by micamu** [comment]
Aren't there some issues that @cdcapano has to fix (Codacy and Travis)?

**Comment by spxiwh** [comment]
This would need to be rebased against master.

**Comment by cdcapano** [comment]
@ahnitz I know you already approved this, but since it's been awhile, just want to double check if it's ok for me to merge.

**Comment by ahnitz** [comment]
@cdcapano Well, it would be nice to clean up a couple of the latter codeclimate issues. Also @micamu should approve.

**Comment by cdcapano** [comment]
@micamu I've updated to master... I don't think there's much I can do about the codeclimate issues. Some of the "similar code" complaints seem a little too stringent.

**Comment by cdcapano** [comment]
The recent changes to `conversions` keeps breaking this... I'm putting on hold until #2291 is merged so I don't have to rebase again again. Also, #2291 would be useful for this patch.

**Comment by cdcapano** [comment]
@micamu Ok, this is really ready now... I've vectorized the functions, so they should work both in the PE code, and for plotting.

File: pycbc/conversions.py [hunk 1]
```diff
@@ -764,23 +764,236 @@ def snr_from_loglr(loglr):
...
+        The damping time of the QNM(s), in seconds. If only a single mode is
+        requested (and mass, spin, l, and m are not arrays), this will be a
+        float. If multiple modes requested, will be an array with shape
+        ``[input shape x] nmodes``, where ``input shape`` is the broadcasted
+        shape of the inputs.
+    """
+    return get_lm_f0tau(final_mass, final_spin, l, m, nmodes)[1]
+
+
+# following are from Berti et al. 2006.
```
Comment (micamu) [inline pycbc/conversions.py]: Could we put the arxiv number here? Berti et al. have a few papers from 2006 and it would be easier to find which one (especially if we want to extend this to higher order modes some day).

File: pycbc/conversions.py [hunk 2]
```diff
@@ -1007,7 +1220,8 @@ def nltides_gw_phase_diff_isco(f_low, f0, amplitude, n, m1, m2):
            'spin2y_from_mass1_mass2_xi2_phi_a_phi_s',
            'chirp_distance', 'det_tc', 'snr_from_loglr',
            'freq_from_final_mass_spin', 'tau_from_final_mass_spin',
+           'final_spin_from_f0_tau', 'final_mass_from_f0_tau',
            'optimal_dec_from_detector', 'optimal_ra_from_detector',
            'chi_eff_from_spherical', 'chi_p_from_spherical',
-           'nltides_gw_phase_diff_isco'
+           'nltides_gw_phase_diff_isco',
```
Comment (micamu) [inline pycbc/conversions.py]: Why did you add here a comma before the end of the list? Is this needed?

File: pycbc/distributions/qnm.py [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+    def __contains__(self, params):
+        isin = super(UniformF0Tau, self).__contains__(params)
+        if isin:
+            isin &= self._constraints(params)
+        return isin
+
+    def _constraints(self, params):
+        f0 = params[self.rdfreq]
+        tau = params[self.damping_time]
+        # temporarily silence invalid warnings... these will just be ruled out
```
Comment (micamu) [inline pycbc/distributions/qnm.py]: This will get rid of all those "invalid value encountered in [...]" warning, is that correct? I'd love that!!

File: pycbc/waveform/ringdown.py [hunk 4]
```diff
@@ -29,6 +29,7 @@
 import lalsimulation as lalsim
 from pycbc.types import TimeSeries, FrequencySeries, float64, complex128, zeros
 from pycbc.waveform.waveform import get_obj_attrs
+from pycbc.conversions import get_lm_f0tau
```
Comment (micamu) [inline pycbc/waveform/ringdown.py]: If this function is only going to be in the conversions module now, then the function `get_lm_f0tau_allmodes` should also be in conversions, or this function could be generalized to have the ability to return various (l, m) modes... This is not pressing though, it could be done in a separate PR later, but if that's the case we should open an issue so that we don't forget to do that in the future.

File: pycbc/conversions.py (hunk 1 above)
Comment (cdcapano) [reply pycbc/conversions.py]: @micamu Ok, I added the reference.

File: pycbc/conversions.py (hunk 2 above)
Comment (cdcapano) [reply pycbc/conversions.py]: This is so that you don't get a merge conflict every time someone adds a new function to the list.

File: pycbc/distributions/qnm.py (hunk 3 above)
Comment (cdcapano) [reply pycbc/distributions/qnm.py]: I think it will... not sure.
